lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -249,6 +251,37 @@ def add_payment_method(post, payment_method)
         post[:b3] = format(payment_method.month, :two_digits)
       end
 
+      def add_stored_credential(post, options)
+        return unless (stored_credential = options[:stored_credential])
+        if stored_credential[:initial_transaction]
+          add_stored_credential_initial(post, options)
+        else
+          add_stored_credential_used(post, options)
+        end
+      end
+
+      def add_stored_credential_initial(post, options)
+        case options[:stored_credential][:reason_type]
+        when 'recurring'
+          post[:a9] = '1' if options[:stored_credential][:initiator] == 'merchant'
+        when 'installment'
+          post[:a9] = '8' if options[:stored_credential][:initiator] == 'merchant'
+        when 'unscheduled'
+          options[:stored_credential][:initiator] == 'merchant' ? post[:a9] = '8' : post[:a9] = '9'
+        end
+      end
+
+      def add_stored_credential_used(post, options)
+        case options[:stored_credential][:reason_type]
+        when 'recurring'
+          post[:a9] = '2' if options[:stored_credential][:initiator] == 'merchant'
+        when 'installment'
+          post[:a9] = '8' if options[:stored_credential][:initiator] == 'merchant'
+        when 'unscheduled'
+          options[:stored_credential][:initiator] == 'merchant' ? post[:a9] = '8' : post[:a9] = '9'
+        end
+      end
```

Comment:
I've made this more DRY as suggested. I was overthinking this by leaving place holders in the future case that Credorax does start supporting 'installment' transactions. I realize this could be easily rewritten in its updated form, if they do start supporting 'installment'.

Reply:
I also think it would be best to pull the `transaction_type` override into this method as well, instead of counting on it getting assigned and then reassigned after when `add_transaction_type` is called outside of this method.

Reply:
Okay, I updated this method to include `post[:a9] = options[:transaction_type] if options[:transaction_type]` after the 'if' statement. Is that what you meant?

Reply:
I pulled the `add_transaction_type` method into the `add_stored_credential` method, so that `:a9` is only set once when `purchase` or `authorize` is called. Let me know what you think. Thanks!